### PR TITLE
Generally improved codebase (per #22), and small fixes.

### PR DESCRIPTION
**Pull request by synchronizing**
As discussed in #22, this is a codebase restructure for Python side of the project. Other things done:

* Added new `errors.py` that contains custom errors.
* Expanded on error returns to the client (including network error, like DDoS prevention).
* Added check to ensure pulled historical data is accurate and conforms the given start/end time by 1%.
* Added new test, and sped up Travis-CI build from 3m to 1m by changing historical data test's gathering range from 1m interval, 5 day range; to 1h interval, 1 day range.

### CODE REVIEW

**Comment by synchronizing** [comment]
After merge I will work on #34 to ensure codebase is working concurrently with Frontman.

File: CryptoBook-py/CryptoBook/handlers.py [hunk 1]
```diff
@@ -127,19 +148,28 @@
...
                 "symbols": ex.symbols,
             },
+            400,
         )
 
     # Closing connection with the market.
     await ex.close()
 
+    # Attempts to get the request.
+    try:
```
Comment (Alescontrela) [inline CryptoBook-py/CryptoBook/handlers.py]: I see that you use a similar error handling statement above in `ip(...)`. Perhaps consider grouping the error handling statements in `ip()` and `historical_data()` into a single `try_request(...)` method a la: 

```
def try_request(func, *args, **kwargs):
  try:
    resp = await func(*args, **kwargs)
  except ExchangeDataAccuracyError as e:
    return {"error": "exchange_data_accuracy_error", "description": str(e)}, 400
  except NetworkError as e:  # pragma: no cover
    return {"error": "network_error", "description": str(e)}, 400
  except Exception as e:  # pragma: no cover
    return {"error": "server_error", "description": str(e)}, 400 
```
so `ip()` now invokes:
```
resp = await try_request(get_exchange_info, exchange=exchange)
```

and `historical_data()` now invokes:
```
resp = await try_request(get_historical_data, **request)
```

Comment (synchronizing) [reply CryptoBook-py/CryptoBook/handlers.py]: I like this idea. Will change the codebase!

Comment (synchronizing) [reply CryptoBook-py/CryptoBook/handlers.py]: Added on [cb8ad5a](https://github.com/Waultics/CryptoBook/pull/30/commits/cb8ad5a366710fa63099b35755b2b53123c91112). Might have to add a few tests for coverage not dip under 100%.

Comment (synchronizing) [reply CryptoBook-py/CryptoBook/handlers.py]: Removed coverage on function `util:handle` on [23b2895](https://github.com/Waultics/CryptoBook/pull/30/commits/23b289592042849c68610279d6adc9cf97f8a24e).

**Pull request merged**